### PR TITLE
Adds customization options for API domain and image lambda

### PR DESCRIPTION
**Pull request by fabiob**
Hi! Thanks for this great tool, it's helping me a lot. Here's my first contribution, I hope you like it.

### Customization of image lambda memory and timeout parameters

My team is currently working on a website in which the users upload very large images, and the image handler is having a hard time optimizing them with the default settings. I've found the perfect settings for my case (60s timeout and 1Gb memory), but I'm pretty sure this should not be the default. Hence my contribution, to allow their customization (both by CLI and the new `CdkUtils` class).

To avoid repeating the defaults over and over, I've established some constants on the `imageLambda.ts` file with the default values. Let me know if this is OK.

### Customization of the API domain

While the Next.js rewrite support works nicely, it increases the total cost of the deployment since the server API will be charged while the real backend does its work. This gets worse when the backend is also deployed as a lambda function on the same account, as they will consume two concurrent execution quotas.

The new option should allow full customization of the `/api` origin when using the `CdkUtils` class, but can also create a simple `HttpOrigin` with an specified domain using the CLI.

## Notes

* I've noticed there's a `.prettierrc`, but most files are not respecting the configured formatting. I tried to minimize the diff, so I reverted any big changes prettier introduced. Let me know if that's ok.
* I haven't updated the package version or the `CHANGELOG.md`, should I?

### CODE REVIEW

File: lib/cdk/utils/imageLambda.ts [hunk 1]
```diff
@@ -8,9 +8,14 @@ export interface SetupImageLambdaProps {
 	assetsBucket: Bucket
 	layerPath: string
 	lambdaHash: string
+	memory?: number
```
Comment (sladg) [inline lib/cdk/utils/imageLambda.ts:11]: I'm thinking if we should go with optional parameters or enforce them.
Personally my preference would be to enforce, especially considering the logic in parent that uses `proces.env` OR default. Open to discussion

**Comment by sladg** [comment]
Perfect! Thank you very much @fabiob! 

Changes makes total sense and implementation is flawless to what I would do myself.

To address your notes:
- yeah, I'm gonna need to improve the contribution guide as well as eslint/prettier config a bit, my focus was on features and it "worked on my localhost" 😬 
- version bump and CHANGELOG will get updated once merged into master :) no need to worry about that

File: lib/cdk/app.ts [hunk 2]
```diff
@@ -30,8 +34,11 @@ new NextStandaloneStack(app, process.env.STACK_NAME, {
 
 	lambdaTimeout: process.env.LAMBDA_TIMEOUT ? Number(process.env.LAMBDA_TIMEOUT) : 15,
```
Comment (sladg) [inline lib/cdk/app.ts:35]: I'm thinking that same should be done for server-lambda as well, in terms of defining constant, I will put a note for myself

File: lib/cdk/utils/imageLambda.ts (hunk 1 above)
Comment (fabiob) [reply lib/cdk/utils/imageLambda.ts:11]: I would vote for enforcing if the only entrypoint were the CLI. With the introduction of the `CdkUtils` class, I think the "batteries included but replaceable" approach is better.

File: lib/cdk/app.ts (hunk 2 above)
Comment (fabiob) [reply lib/cdk/app.ts:35]: I guess so… But I tried to minimize the diff for now.

(I wish I could say the same for my imaginex PR :stuck_out_tongue_closed_eyes: )

**Pull request merged**